libs/pages/services/src/lib/ui/page-application-create/page-application-create-general/page-application-create-general.tsx

```diff
@@ -0,0 +1,109 @@
+import { FormEventHandler } from 'react'
+import { Controller, useFormContext } from 'react-hook-form'
+import { useNavigate, useParams } from 'react-router'
+import { IconEnum, ServiceTypeEnum } from '@qovery/shared/enums'
+import { OrganizationEntity } from '@qovery/shared/interfaces'
+import { SERVICES_URL } from '@qovery/shared/router'
+import { Button, ButtonSize, ButtonStyle, InputSelect, InputText, Link } from '@qovery/shared/ui'
+import { GlobalData } from '../../../feature/page-application-create-feature/interfaces.interface'
+import PageApplicationCreateGeneralContainer from './page-application-create-general-container/page-application-create-general-container'
+import PageApplicationCreateGeneralGitApplication from './page-application-create-general-git-application/page-application-create-general-git-application'
+
+export interface PageApplicationCreateGeneralProps {
+  onSubmit: FormEventHandler<HTMLFormElement>
+  organization?: OrganizationEntity
+}
+
+export function PageApplicationCreateGeneral(props: PageApplicationCreateGeneralProps) {
+  const { control, getValues, watch, formState } = useFormContext<GlobalData>()
+  const { organizationId = '', environmentId = '', projectId = '' } = useParams()
+  const navigate = useNavigate()
+
+  watch('applicationSource')
```

Comment:
Useless ?

Reply:
Not useless, if I don't put that, the view is not updated and the selection of either one or the other does not retrigger a render of the view... I was surprised by this behavior but I really needed this peace of code